### ISSUE TITLE
Bug: dispose hooks not called with `using` keyword

### ISSUE THREAD

**Issue by marvinhagemeister** [swc-project/swc#9576]
### Describe the bug

Investigation https://github.com/denoland/deno/issues/25757 lead to realizing that the transpilation of the `using` keyword in SWC doesn't seem to call the dispose hooks when inside a loop. Running the reproduction snippet should log that all dispose methods are called, but they're never called with the transpiled variant.

### Input code

```typescript
class Disposable {
  disposed = false;

  [Symbol.dispose]() {
      this.disposed = true
  }
}

const disposables = [new Disposable()]

for (using _ of disposables) {/* ... */}

if (disposables[0].disposed) {
  console.log("✅ dispose ok")
} else {
  console.error("💥 failed to dispose")
}

class AsyncDisposable {
  disposed = false;

  [Symbol.asyncDispose]() {
      this.disposed = true
  }
}

const asyncDisposables = [new AsyncDisposable()]

for (await using _ of asyncDisposables) {/* ... */}

if (asyncDisposables[0].disposed) {
  console.log("✅ async dispose ok")
} else {
  console.error("💥 failed to async dispose")
}
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": true
    },
    "target": "es2022",
    "loose": false,
    "minify": {
      "compress": false,
      "mangle": false
    },
    "transform": {
      "react": {
        "importSource": "npm:react",
        "runtime": "automatic"
      }
    }
  },
  "module": {
    "type": "es6"
  },
  "minify": false,
  "isModule": true
}
```

### Playground link (or link to the minimal reproduction)

https://play.swc.rs/?version=1.7.26&code=H4sIAAAAAAAAA52QPW4CMRCFe59itJWXwtAjiki5AeVqFZnFS6w4O5HHCCG0JRegRkqVi%2BUEOQKzDj%2FeFUWIO%2FvNm%2Bf3VU4TwbOlDyS9cAZ2AmAZr2YJM6i1IzMV%2FFjMt%2B8LdOosljKPs90Jr5ZUYgp%2BbVhqRStEhQ2F88YugFgvGrNJMmVeClGjB7km26zgBbBOHRw0HoFSCkZj3mhrkIlaTMpr9u%2BXukh0Rjlcyez7uL%2F0AXzLctGC4Uq9QeM9epn9fB6%2BuLB1XCLgxdVZuEbk9ETbpnoElr4ZHiSm%2B1FXbIMv3NjpjbYBEoLDDXcwDkf%2BwDJa%2Fkm0541cT19Nfi1%2FAgAA&config=H4sIAAAAAAAAA1WQPQ7CMAyFd05ReWZAHRi4AxMnsIJbBTU%2Fsh2JCvXuJOn%2FEiXfe8%2BO%2Fbs0DXzEwKP55Wt%2BRGQh3t6ZyOgVv5mAjpHEsI0K11VVKZJyokqmWQBF7klLiKS9te0SgCEEoYw7HIQW5qy33XhsaYKLTCJnY7Gi74ctf%2B7I6KUL7I6FmNDoAWRkXQysr5DYlELgo3vMtutu4uTVuqpj0uBQrYFFni7rWVuDC%2B9U%2F7RssGxpHvwOu2mdcZsHrDzXZF3f9Ae8XWptjQEAAA%3D%3D

### Expected behavior

The reproduction snippet should log out that the dispose hooks are called.

### Version

1.7.26

**Comment by kdy1** [comment]
Status update: I tried mimicking the output of babel, but babel still fails to dispose it. https://babeljs.io/repl#?browsers=&build=&builtIns=false&corejs=false&spec=false&loose=false&code_lz=MYGwhgzhAEAiCWEAOB7CYBGICm0DeAUNNACaKoTYnQC80AZmCJQNwFHQDaAygJ4C2GFCAB0ZZGmwBdABQBKfB2LEALgAtEY8pOp0VAJwCu2DgF8C5gsBQA7CCtLb0WbDDqcb2AO5wnmHPJS7PQo-tAyhhDwNgDm0AD60Cj0jhLOOBAKeAD0AFTQIoXQudmW8Cky4hT-rpwADFJaaVRZHNZ2wtgiICgxMgBEgKDkqRS4KADW_XIW0NjMuITE7RCdItj6-qEDgLwbgKU7DGDwONQqKCOSUxbsoJAwAIIQvDbACGk1isRVOrQH82wcPAEQlEkCeLyc0nkH2Uqg0ECao100AMxjMVystns0FBz1e1Rcbi4nh8DzBeLQNUCwVC4TAXkODki0TiiWS2MeuL8BKyeQKRRKZQqOPBbwJ9UaX0oJFaS0xqx6fSG7LB50oSUm01Ms3m0OWq3Wm30O32jCOVGRZ2Fquwl1MQA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=stage-3&prettier=false&targets=&version=7.25.8&externalPlugins=&assumptions=%7B%7D

**Comment by jtbandes** [comment]
Was this actually fixed? I am still seeing issues in 1.7.39: https://github.com/swc-project/swc/issues/9673
